SSA/Projects/InstCombine/ForMathlib.lean

```diff
@@ -0,0 +1,30 @@
+import Mathlib.Tactic.Ring
+import Std.Data.BitVec
+import Mathlib.Data.BitVec.Lemmas
+import Mathlib.Data.BitVec.Defs
+
+namespace Std.BitVec
+open Nat
+
+lemma natCast_eq (x w : Nat) : Nat.cast x = x#w       := rfl
+
+theorem ofFin_intCast (z : ℤ) : ofFin (z : Fin (2^w)) = Int.cast z := by
+  cases w
+  case zero =>
+    simp only [eq_nil]
+  case succ w =>
+    simp only [Int.cast, IntCast.intCast, BitVec.ofInt]
+    unfold Int.castDef
+    cases' z with z z
+    · rfl
+    · simp only [cast_add, cast_one, neg_add_rev]
+      rw [← add_ofFin, ofFin_neg, ofFin_ofNat, ofNat_eq_ofNat, ofFin_neg, ofFin_natCast,
+        natCast_eq, negOne_eq_allOnes, ← sub_toAdd, allOnes_sub_eq_not]
+
+theorem toFin_intCast (z : ℤ) : toFin (z : BitVec w) = z := by
```

Comment:
would something like `@[norm_cast]` make sense here?

Reply:
I have no idea. How would that work? Why would that be preferred?

Reply:
this is a very minor comment/question, mostly interesting later for upstreaming. `norm_cast` is a tactic that handles all of these coercions (see https://lean-forward.github.io/norm_cast/norm_cast.pdf) and the tag is like the `@[simp]` tag to tell the tactic which lemmas to use for these coercions.

Reply:
To be honest, I just copied this from alex Mathlib pr without thinking too much about it. I guess we can refine this in the upstreaming process.